client/api.go

```diff
@@ -167,7 +188,7 @@ func (c *Client) DeleteMessages(messages *Messages, seek *int) error {
 			if msg.Hit {
 				// The message might be an action rather than text. Actions aren't deletable.
 				// An example of an action is a call request.
-				if msg.Type == UserMessage {
+				if msg.Type == UserMessage && !c.SkipChannel(msg.ChannelID) {
```

Comment:
On the right track. With your checks above this one won't be necessary anymore.

Reply:
Near line 112, it resolves relationships which seems to be related to DMs only, the `"channels"` endpoint is for `"channels": "/users/@me/channels"`, i have added a check there now.

`if msg.Type == UserMessage && !c.SkipChannel(msg.ChannelID)` this check is required because i dont think there's any way to skip fetching messages from individual channels in guilds unless the whole way in which guild messages are fetched is changed to something like fetching all guild channel IDs first, and then requesting messages from those IDs. Without this check, only complete guilds or private messages can be skipped, not individual channels in guilds :/

Reply:
Aha sorry yes, I get it now. There is a way we can skip individual channels in guilds.

The code currently makes requests to `/channels/%v/messages/search` to search message history. Discord's API allows filtering (inclusive) by channel ID by appending the query parameter `&channel_id=1234...`. In fact, this query parameter can be specified multiple times to include multiple channels.

We could do this:
* Get list of channel IDs in a guild
* Filter out the channel IDs we want to skip
* Append together the list of remaining channel IDs into `&channel_id=` query parameters and make our message query

What do you think? Up for it? If not, I can take a look at some point.

Reply:
I'll **try** doing it

Reply:
Pushed another commit, untested. Forgot that i had to `Append together the list of remaining channel IDs into &channel_id= query parameters and make our message query` instead of individually going through guild channels, will fix tomorrow.